src/tests/common/control_topic/xi_cbor_codec_ct_server.c

```diff
@@ -0,0 +1,336 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <xi_cbor_codec_ct_server.h>
+#include <xi_cbor_codec_ct_keys.h>
+
+#include <cbor.h>
+#include <cn-cbor/cn-cbor.h>
+
+#include <xively_error.h>
+#include <xi_macros.h>
+#include <xi_debug.h>
+
+#ifdef USE_CBOR_CONTEXT
+
+extern cn_cbor_context* context;
+
+#endif
+
+/* reusing the put_name_and_revision function from libxively here in the mock broker */
+void xi_cbor_put_name_and_revision( cn_cbor* cb_map,
+                                    const char* name,
+                                    const char* revision,
+                                    cn_cbor_errback* errp );
+
+void xi_cbor_codec_ct_server_encode( const xi_control_message_t* control_message,
+                                     uint8_t** out_encoded_allocated_inside,
+                                     uint32_t* out_len )
+{
+    if ( NULL == control_message )
+    {
+        *out_len = 0;
+        return;
+    }
+
+    cn_cbor_errback err;
+    cn_cbor* cb_map = cn_cbor_map_create( CBOR_CONTEXT_PARAM_COMA & err );
+
+    cn_cbor_map_put(
+        cb_map,
+        cn_cbor_string_create( XI_CBOR_CODEC_CT_STRING_MSGTYPE CBOR_CONTEXT_PARAM, &err ),
+        cn_cbor_int_create( control_message->common.msgtype CBOR_CONTEXT_PARAM, &err ),
+        &err );
+
+    cn_cbor_map_put(
+        cb_map,
+        cn_cbor_string_create( XI_CBOR_CODEC_CT_STRING_MSGVER CBOR_CONTEXT_PARAM, &err ),
+        cn_cbor_int_create( control_message->common.msgver CBOR_CONTEXT_PARAM, &err ),
+        &err );
+
+    switch ( control_message->common.msgtype )
+    {
+        case XI_CONTROL_MESSAGE_SC__SFT_FILE_UPDATE_AVAILABLE:
+
+            if ( 0 < control_message->file_update_available.list_len )
+            {
+                cn_cbor* files = cn_cbor_array_create( CBOR_CONTEXT_PARAM_COMA & err );
```

Comment:
it should but still - as we discussed this in previous PR - the code formatter does this with the code.